docs/schema-changes.md

```diff
@@ -0,0 +1,119 @@
+# Schema Changes Data Model
+
+## Introduction
+
+Weaver can be used to compare two versions of a semantic convention registry
+and generate a diff report. This document describes the data model used to
+represent the differences between two versions of a semantic convention
+registry, as well as the diffing process. This diff report can be used to:
+
+- Understand the changes between two versions of a semantic convention registry.
+- Update the OpenTelemetry Schema file, section versions, with the new version.
+- Generate a migration guide for users of the semantic convention registry.
+- Generate a SQL DDL script to update a database schema.
+- And more.
+
+> Note: The current implementation of the diffing process focuses on the top-level
+items attributes, metrics, events, spans, resources and does not compare the fields
+of those top-level items.
+
+## Data Model
+
+The schema changes data model is composed of the following components:
+
+- `head`: The registry manifest of the most recent version of the semantic
+  convention registry used in the diffing process.
+- `baseline`: The registry manifest of the oldest version of the semantic
+  convention registry used in the diffing process.
+- `changes`: A dictionary of changes between the head and baseline registries.
+  The dictionary is composed of the following keys (when applicable):
+  - `attributes`: A list of changes to attributes.
+  - `metrics`: A list of changes to metrics.
+  - `events`: A list of changes to events.
+  - `spans`: A list of changes to spans.
+  - `resources`: A list of changes to resources.
+
+Each change in the changes dictionary for any key is represented as a list of
+schema changes, represented by one of the following types:
+
+- `added`: A new schema item (e.g., attribute, metric, etc.) was added in the
+  head registry. The name of the new item is stored in the name attribute.
+- `renamed_to_new`: One or more schema items in the baseline registry were
+  renamed to the same new name in the head registry. The old names of the
+  items are stored in the old_names attribute, and the new name is stored in
+  the current_name attribute.
+- `renamed_to_existing`: One or more schema items in the baseline registry were
+  renamed to an existing item in the head registry. The old names of the items
+  are stored in the old_names attribute, and the existing item name is stored
+  in the current_name attribute.
+- `deprecated`: An item in the baseline registry was deprecated in the head
+  registry. The name of the deprecated item is stored in the name attribute,
+  and the deprecation note is stored in the note attribute.
+- `removed`: An item in the baseline registry was removed in the head
+  registry. The name of the removed item is stored in the name attribute.
+
```

Comment:
we'll eventually need a way to describe other (breaking) changes such as attribute type, metric unit or instrument change. 

Maybe we can add `other` category that would be used for change types that we don't formally support yet? This would allow someone to look at the diff and understand that there was some change and maybe some manual intervention is necessary. 
Otherwise, the lack of any mention could be perceived as 'there was no change, all looks good and compatible'.

Reply:
I will add a note and describe how this could be represented as a potential future extension.

Reply:
Done